src/libraries/System.Private.CoreLib/src/System/Numerics/Vector.cs

```diff
@@ -1431,5 +1433,27 @@ internal static void ThrowInsufficientNumberOfElementsException(int requiredElem
         {
             throw new IndexOutOfRangeException(SR.Format(SR.Arg_InsufficientNumberOfElements, requiredElementCount, "values"));
         }
+
+        /// <summary>
+        /// Reinterprets a <see cref="Vector{T}"/> as a <see cref="Vector{T}"/> of new type.
+        /// </summary>
+        /// <typeparam name="TFrom">The type of the input vector.</typeparam>
+        /// <typeparam name="TTo">The type to reinterpret the vector as.</typeparam>
+        /// <param name="vector">The vector to reinterpret.</param>
+        /// <returns><paramref name="vector"/> reinterpreted as a new <see cref="Vector{T}"/>.</returns>
+        /// <exception cref="NotSupportedException">
+        /// The type of <typeparamref name="TFrom"/> or <typeparamref name="TTo"/> is not supported.
+        /// </exception>
+        public static Vector<TTo> As<TFrom, TTo>(Vector<TFrom> vector)
+            where TFrom : struct
+            where TTo : struct
+        {
+            ThrowHelper.ThrowForUnsupportedVectorBaseType<TFrom>();
+            ThrowHelper.ThrowForUnsupportedVectorBaseType<TTo>();
+
+            Debug.Assert(Vector<TFrom>.Count * Unsafe.SizeOf<TFrom>() == Vector<TTo>.Count * Unsafe.SizeOf<TTo>());
```

Comment:
I don' think this assert is necessary. If the size, in bytes, of `Vector<T>` vs `Vector<U>` varies then we have much bigger problems in the JIT 😄